File: leetcode/1647/MinDeletions.cs
public class Solution {
    public int MinDeletions(string s) {
        int[] freq = new int[26];
        
        // char[] arr;
        // arr = s.ToCharArray(0, s.Length);
        
        foreach (char c in s.ToCharArray(0, s.Length)) {
            freq[c - 'a']++;
        }
		Array.Sort(freq);
        int exp = freq[25];
        int res = 0;
        for (int i = 25; i >= 0; i--) {
            if (freq[i] == 0) break;
            if (freq[i] > exp) {
                res += freq[i] - exp;
            } else {
                exp = freq[i];
            }
            if (exp > 0) exp--; // Lowest exp is zero, cannot be negative
        }
        return res;
    }
}

File: leetcode/1647/MinDeletions.js
/**
 * @param {string} s
 * @return {number}
 */
var minDeletions = function(s) {
    
    let freq = [...new Array(26)].map(()=>0);
    
    let arr = s.split('');
    for(let i = 0; i < arr.length; i++)
    {
        freq[arr[i].charCodeAt() - 'a'.charCodeAt()]++;
    }
    
    // console.log(freq);
    freq.sort((a,b)=>a-b);
    // freq.sort();
    let exp = freq[25];
    let res = 0;
    
    for(let i = 25; i >-1; i--)
    {
        if( freq[i] == 0 )
            break;
        
        if(freq[i] > exp)
        {
            res += freq[i] - exp;
        }else
        {
            exp = freq[i];        
        }
        
        if(exp > 0)
            exp--;
    }
    return res;
};

console.log(minDeletions("aaabbbcc"));